sockeye/layers.py

```diff
@@ -225,6 +228,85 @@ def __call__(self,
                                     bias=bias,
                                     flatten=False)
 
+class LengthRatioConfig(config.Config):
+    """
+    Configuration of the length ratio predictor.
+
+    :param layers: Number of layers.
+    :param weight: Weight of this loss.
+    """
+
+    def __init__(self, num_layers: int, weight: float) -> None:
+        super().__init__()
+        self.num_layers = num_layers
+        # TODO: keep weight here redundant because the weight is also stored in the loss config.
+        # but it's used to test if we need length prediction at all
+        self.weight = weight
+
+
+class LengthRatio:
+    """
+    Defines the length-ratio prediction layer of Sockeye.
+
+    :param hidden_size: Decoder hidden size.
+    :param prefix: Prefix used for naming.
+    """
+
+    def __init__(self,
+                 hidden_size: int,
+                 num_layers: int,
+                 weight_normalization: bool,
+                 prefix: str = C.LENRATIOS_OUTPUT_LAYER_PREFIX,
+                 name: str = C.LENRATIO_OUTPUT_NAME) -> None:
+        utils.check_condition(num_layers >= 1, "LengthRatio's num_layers has to be >=1.")
+        self.prefix = prefix
+        self.name = name
+        self.num_layers = num_layers
+
+        self.layers = []
+        # first the internal layers
+        for l in range(self.num_layers - 1):
+            self.layers.append(OutputLayer(hidden_size=hidden_size,
```

Comment:
I guess this works fine, but I find it a bit confusing to use OutputLayer here.
You could just as well do the following in ___call__():
```python
data
for l in range(self.num_layers - 1):
  data = mx.sym.FullyConnected(data=data, num_hidden=hidden_size, flatten=False, name=self.name + 'fc%d' %l)
data = mx.sym.FullyConnected(data=data, num_hidden=1, flatten=false ...)
```